### PR TITLE
cli: Fix php version check

### PR DESCRIPTION
**Pull request by anomiex**
## Proposed changes:

Some of the sury packages use versions like 8.3.3-1 in `PHP_VERSION` when they add patches, which confuses `semver.satisfies`. Strip those off.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1710432839585779/1710425419.002989-slack-C05Q5HSS013

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Spin up an Ubuntu docker container, install the 8.3 package from https://deb.sury.org/ (which currently has version "8.3.3-1+ubuntu22.04.1+deb.sury.org+1"), then see if running `jetpack` gives you a bogus warning.

### CODE REVIEW

File: tools/cli/helpers/checkEnvironment.js [hunk 1]
```diff
@@ -108,7 +108,8 @@ export async function getPhpVersion() {
 		}
 		return;
 	}
-	return res.stdout.toString().trim();
+	// Strip any `-1` or the like, as various sources (e.g. sury) indicate patched versions in this manner which confuses `semver.satisfies`.
+	return res.stdout.toString().trim().replace( /-\d.*/, '' );
```
Comment (tbradsha) [inline tools/cli/helpers/checkEnvironment.js]: I'd lean toward something more general:
```suggestion
	return res.stdout.toString().trim().replace( /-.*/, '' );
```

For example, I looked through some logs and found that `8.1.22-he.0` is a valid version, as is `8.3.0-dev`.

Comment (anomiex) [reply tools/cli/helpers/checkEnvironment.js]: `8.3.0-dev` is a prerelease version, I was unsure about whether we should mess with that.

I don't know what `8.1.22-he.0` might be for.

Comment (tbradsha) [reply tools/cli/helpers/checkEnvironment.js]: If the ideal is 8.2, but we allow 8.3, we should also allow 8.3-dev in my opinion.

As for `8.1.22-he.0` (and other ones with the `he.0` suffix), I'm not sure what it's from, but it was common in some logs I was perusing. It's a valid version, though, the same as any other dash suffix, so if we're going to allow some dash suffixes, why not all?

Comment (anomiex) [reply tools/cli/helpers/checkEnvironment.js]: Yeah, I think you're right.

**Review by tbradsha — APPROVED** [review]

**Pull request merged**